code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -218,30 +225,63 @@
 	target.hotspot_expose((ptank.air_contents.temperature*2) + 380,500)
 	//location.hotspot_expose(1000,500,1)
 	SSair.add_to_active(target, 0)
-	return
 
 
-/obj/item/weapon/flamethrower/full/New(var/loc)
-	..()
-	if(!weldtool)
-		weldtool = new /obj/item/weapon/weldingtool(src)
-	weldtool.status = 0
-	if(!igniter)
-		igniter = new /obj/item/device/assembly/igniter(src)
-	igniter.secured = 0
-	status = 1
-	update_icon()
+/obj/item/weapon/flamethrower/Initialize(mapload)
+	. = ..()
+	if(create_full)
+		if(!weldtool)
+			weldtool = new /obj/item/weapon/weldingtool(src)
+		weldtool.status = 0
+		if(!igniter)
+			igniter = new igniter_type(src)
+		igniter.secured = 0
+		status = 1
+		if(create_with_tank)
+			ptank = new /obj/item/weapon/tank/internals/plasma/full(src)
+		update_icon()
+
+/obj/item/weapon/flamethrower/proc/create_igniter()
 
-/obj/item/weapon/flamethrower/full/tank/New(var/loc)
-	..()
-	ptank = new /obj/item/weapon/tank/internals/plasma/full(src)
-	update_icon()
 
+/obj/item/weapon/flamethrower/full/tank
+	create_full = TRUE
+
+/obj/item/weapon/flamethrower/full/tank
+	create_with_tank = TRUE
 
 /obj/item/weapon/flamethrower/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance, damage, attack_type)
 	if(ptank && damage && attack_type == PROJECTILE_ATTACK && prob(15))
 		owner.visible_message("<span class='danger'>[attack_text] hits the fueltank on [owner]'s [src], rupturing it! What a shot!</span>")
 		var/target_turf = get_turf(owner)
-		ignite_turf(target_turf, 100)
+		igniter.ignite_turf(src,target_turf, release_amount = 100)
 		qdel(ptank)
 		return 1 //It hit the flamethrower, not them
+
+
+/obj/item/weapon/flamethrower/freeze
+	name = "freezethrower"
+	desc = "Is it getting cold in here?"
+	igniter_type = /obj/item/device/assembly/igniter/cold
+
+/obj/item/weapon/flamethrower/freeze/full
+	create_full = TRUE
+
+/obj/item/weapon/flamethrower/freeze/full/tank
+	create_with_tank = TRUE
+
+/obj/item/device/assembly/igniter/cold/ignite_turf(obj/item/weapon/flamethrower/F,turf/open/location,release_amount = 0.05)
+	if(!F.ptank || !istype(location))
+		return
+	var/datum/gas_mixture/air_transfer = F.ptank.air_contents.remove_ratio(release_amount)
+	if(air_transfer.gases["plasma"])
```

Comment:
Why is it plasma

Why do you get to bitch about UGH FREEON SO UNREALISTIC

then immediately turn around and make Plasma do it